extensions/SwiftCloudFiles/php-cloudfiles-1.7.10/cloudfiles_http.php

```diff
@@ -240,7 +240,43 @@ public function authenticate( $user, $pass, $host = NULL ) {
 		return array( $this->response_status, $this->response_reason,
 			$this->storage_url, $this->cdnm_url, $this->auth_token );
 	}
-	
+
+	// Wikia change - begin
+	// retry auth request in case of an error (PLATFORM-1659)
+	public function authenticate( $user, $pass, $host = NULL ) {
+		$retriesLeft = self::MAX_RETRIES;
+		$res = false;
+
+		wfDebug( __METHOD__ . "\n" );
+
+		while( $retriesLeft >= 0 ) {
+			list( $status,$reason,$surl,$curl,$atoken ) = $this->do_authenticate( $user, $pass, $host );
+
+			# PLATFORM-1659 - retry all HTTP 50x responses
+			if ( ($status >= 200 && $status <= 299) ) {
+				wfDebug( __METHOD__ . ": ok\n" );
+				break;
+			}
+
+			# PLATFORM-1521 - report an error only when there are no retries left
+			$level = ( $retriesLeft === 0 ) ? 'error' : 'warning';
+
+			Wikia\Logger\WikiaLogger::instance()->$level( 'SwiftStorage: authentication retry', [
+				'exception'    => new SwiftRetryException( $reason, is_numeric($status) ? $status : 0 ),
+				'retries-left' => $retriesLeft,
+			] );
+
+			wfDebug( sprintf( "%s : retrying as I got '%s' (%d retries left)\n", __METHOD__ , trim( $reason ), $retriesLeft ) );
+
+			// wait a bit before retrying the request
+			usleep( self::RETRY_DELAY * 1000 );
+			$retriesLeft--;
+		}
+
+		return array( $status,$reason,$surl,$curl,$atoken );
```

Comment:
Whitespacing fixed.

Change the array notation to a longer one to keep the consistency.